source/_integrations/generic_water_heater.markdown

```diff
@@ -0,0 +1,50 @@
+---
+title: Generic Water Heater
+description: Turn Home Assistant into a Water Heater
+ha_category:
+  - Water Heater
+ha_release: 2012.02
+ha_iot_class: Local Polling
+ha_domain: generic
+---
+
+The `Generic Water Heater` water_heater platform is a Water Heater implemented in Home Assistant. It uses a sensor and a switch connected to a domestic hot water (DHW) reservoir and a heat source (eg. resistance). When "on", if the measured temperature is cooler than the target temperature, the heater will be turned on, and turned off when the required temperature is reached. 
+
+```yaml
+# Example configuration.yaml entry
+water_heater:
+  - platform: generic
+    name: bath water
+    heater: switch.dhw_switch
+    target_sensor: sensor.dhw_temperature
+    target_temperature: 50
+    delta_temperature: 5
```

Comment:
This violates ADR-0007

Reply:
https://github.com/home-assistant/example-custom-config/tree/master/custom_components/example_light/ is not longer up to date, right ?

Reply:
refactored

Whats the stance on these generic platforms? should or shouldn't have config flows ?